oe_paragraphs.install

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * @file
+ * Install, update and uninstall functions for the OE Paragraphs module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Config\FileStorage;
+use Drupal\Core\Entity\Entity\EntityFormDisplay;
+use Drupal\Core\Entity\Entity\EntityFormMode;
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Copy the values to the new variant field and delete the old variant fields.
+ */
+function oe_paragraphs_update_8001(array &$sandbox) {
+  if (!isset($sandbox['total'])) {
+    // Initialise batch variables and settings.
+    $sandbox['variant_fields'] = [
+      'oe_content_row' => 'field_oe_content_row_variant',
+      'oe_list_item_block' => 'field_oe_list_item_block_variant',
+      'oe_list_item' => 'field_oe_list_item_variant',
+    ];
+    $paragraph_ids = [];
+
+    foreach ($sandbox['variant_fields'] as $bundle => $field) {
+      // Make sure we have the old fields in the database.
+      $all_bundle_fields = \Drupal::service('entity_field.manager')->getFieldDefinitions('paragraph', $bundle);
+      if (!isset($all_bundle_fields[$field])) {
+        continue;
+      }
+
+      // Take all entities that has a value in one of the old fields.
+      $query = \Drupal::entityQuery('paragraph');
+      $query->exists($field);
+      $result = $query->execute();
+      if (!empty($result)) {
+        $paragraph_ids += $result;
+      }
+
+      $sandbox['total'] = count($paragraph_ids);
+      $sandbox['current'] = 0;
+      $sandbox['paragraphs_per_batch'] = 25;
+    }
+
+    // Make sure we have the new base field created before we copy values.
+    $storage_definition = BaseFieldDefinition::create('string')
+      ->setLabel(t('Variant'))
+      ->setRequired(FALSE)
+      ->setTranslatable(FALSE)
+      ->setRevisionable(TRUE)
+      ->setSetting('max_length', 255)
+      ->setDisplayConfigurable('form', FALSE)
+      ->setDisplayConfigurable('view', FALSE);
+
+    \Drupal::entityDefinitionUpdateManager()
+      ->installFieldStorageDefinition('field_oe_paragraphs_variant', 'paragraph', 'oe_paragraphs', $storage_definition);
+  }
+
+  // Return if we have nothing to process.
+  if ($sandbox['total'] === 0) {
+    \Drupal::messenger()->addMessage($sandbox['current'] . ' paragraphs processed.');
```

Comment:
There is a bit of duplication here (same code as in the "finished" procedure) and also the function will end without running other updates (like setting the form modes). Can't we simply set the finished to 1 and then let the function progress on its own?